src/riak_kv_wm_queue.erl

```diff
@@ -107,9 +198,36 @@ produce_queue_fetch(RD, Ctx) ->
                         riak_client:fetch(QueueName, Client),
                         RD,
                         Ctx).
-    
-format_response(internal, {ok, queue_empty}, RD, Ctx) ->
+
+
+format_response(_, {ok, queue_empty}, RD, Ctx) ->
     {<<0:8/integer>>, RD, Ctx};
+format_response(_, {error, Reason}, RD, Ctx) ->
+    lager:warning("Fetch error ~w", [Reason]),
+    {{error, Reason}, RD, Ctx};
+format_response(internal_aaehash, {ok, {deleted, TombClock, RObj}}, RD, Ctx) ->
+    BK = make_binarykey(riak_object:bucket(RObj), riak_object:key(RObj)),
+    {SegmentID, SegmentHash} =
+        leveled_tictac:tictac_hash(BK, lists:sort(TombClock)),
+    SuccessMark = <<1:8/integer>>,
+    IsTombstone = <<1:8/integer>>,
+    ObjBin = encode_riakobject(RObj),
+    TombClockBin = term_to_binary(TombClock),
+    TCL = byte_size(TombClockBin),
+    {<<SuccessMark/binary, IsTombstone/binary,
```

Comment:
Should this really be part of web machine queue handler? So much detail in building a binary that I can hardly believe this is the right place. Rather have it somewhere in are or so...

But also, would you want this as a binary or rather as a JSON object if you're anyway doing it for having 3rd party apps handle the output.

Reply:
To  certain extent the use of a binary is locked in here, as this is how it is done for nextgenrepl.  I think the right thing to do for the future is to properly support an alternative to the `internal` and `internal_aaehash` encoding options which is to have a `external_json` alternative.